pkg/operation/botanist/component/vpnseedserver/vpn_seed_server.go

```diff
@@ -0,0 +1,662 @@
+						},
+					},
+				},
+				Tls: &v1beta1.ClientTLSSettings{
+					Mode: v1beta1.ClientTLSSettings_DISABLE,
+				},
+			},
+		}
+		return nil
+	}); err != nil {
+		return err
+	}
+	if _, err := controllerutil.CreateOrUpdate(ctx, v.client, virtualService, func() error {
+		virtualService.Spec = v1beta1.VirtualService{
+			ExportTo: []string{"*"},
+			Hosts:    []string{*v.kubeAPIServerHost},
+			Gateways: []string{DeploymentName},
+			Http: []*v1beta1.HTTPRoute{
+				{
+					Route: []*v1beta1.HTTPRouteDestination{
+						{
+							Destination: &v1beta1.Destination{
+								Port: &v1beta1.PortSelector{
+									Number: 1194,
+								},
+								Host: DeploymentName,
+							},
+						},
+					},
+				},
+			},
+		}
+		return nil
+	}); err != nil {
+		return err
+	}
+
+	if _, err := controllerutil.CreateOrUpdate(ctx, v.client, service, func() error {
+		service.Annotations = map[string]string{
+			"networking.istio.io/exportTo": "*",
+		}
+		service.Spec.Type = corev1.ServiceTypeClusterIP
+		service.Spec.Ports = []corev1.ServicePort{
+			{
+				Name:       DeploymentName,
+				Port:       1194,
+				TargetPort: intstr.FromInt(1194),
+			},
+			{
+				Name:       "http-proxy",
+				Port:       9443,
+				TargetPort: intstr.FromInt(9443),
+			},
+		}
+		service.Spec.Selector = map[string]string{
+			"app": DeploymentName,
+		}
+		return nil
+	}); err != nil {
+		return err
+	}
+
+	return nil
+}
+
+func (v *vpnSeedServer) Destroy(ctx context.Context) error {
+	var (
+		serverSecret    = v.emptySecret(DeploymentName)
+		tlsAuthSecret   = v.emptySecret(fmt.Sprintf("%s-tlsauth", DeploymentName))
+		dhSecret        = v.emptySecret(fmt.Sprintf("%s-dh", DeploymentName))
+		networkPolicy   = v.emptyNetworkPolicy()
+		deployment      = v.emptyDeployment()
+		configMap       = v.emptyConfigMap()
+		gateway         = v.emptyGateway()
+		destinationRule = v.emptyDestinationRule()
+		virtualService  = v.emptyVirtualService()
+		service         = v.emptyService()
+	)
+
+	if err := client.IgnoreNotFound(v.client.Delete(ctx, serverSecret)); err != nil {
+		return err
+	}
+	if err := client.IgnoreNotFound(v.client.Delete(ctx, tlsAuthSecret)); err != nil {
+		return err
+	}
+	if err := client.IgnoreNotFound(v.client.Delete(ctx, dhSecret)); err != nil {
+		return err
+	}
+	if err := client.IgnoreNotFound(v.client.Delete(ctx, networkPolicy)); err != nil {
+		return err
+	}
+	if err := client.IgnoreNotFound(v.client.Delete(ctx, deployment)); err != nil {
+		return err
+	}
+	if err := client.IgnoreNotFound(v.client.Delete(ctx, configMap)); err != nil {
+		return err
+	}
+	if err := client.IgnoreNotFound(v.client.Delete(ctx, gateway)); err != nil {
+		return err
+	}
+	if err := client.IgnoreNotFound(v.client.Delete(ctx, destinationRule)); err != nil {
+		return err
+	}
+	if err := client.IgnoreNotFound(v.client.Delete(ctx, virtualService)); err != nil {
+		return err
+	}
+	if err := client.IgnoreNotFound(v.client.Delete(ctx, service)); err != nil {
+		return err
+	}
```

Comment:
Use `kutil.DeleteObjects(...)`

Reply:
Thanks for the hint. That was what we were looking for.